extra/src/main/scala/ai/lum/odinson/extra/AnnotateText.scala

```diff
@@ -66,7 +64,13 @@ object AnnotateText extends App with LazyLogging {
 
   // NOTE parses the documents in parallel
   for (f <- textDir.listFilesByWildcard("*.txt", caseInsensitive = true, recursive = true).par) {
-    val docFile = new File(docsDir, f.getBaseName() + ".json.gz")
+    val relFile = textDir.toPath.relativize(f.toPath)
+
+    // replicate the directory structure under textDir under docsDir
+    val inputFileInDocsDir = docsDir.toPath.resolve(relFile)
+    val docFile = inputFileInDocsDir.getParent
+      .resolve(inputFileInDocsDir.getFileName.toFile.getBaseName() + ".json.gz").toFile
+    Ensuring(docFile.toPath.getParent.toFile.mkdirs)
```

Comment:
If the dir structure already exists, this will fail?
I would think that we would rather use it (instead of failing), unless I'm missing something?

Reply:
Good catch - will add test and fix if need be

Reply:
Confirmed bug and fix by:
1. reading `mkdirs` docs (imagine that, predicting what your assertion will do!?)
2. hacking test to use fixed dir `private val tmpFolder = Files.createDirectory(Paths.get("/tmp/odinson-test/")).toFile.getAbsolutePath` and removing the `deleteDocs()` calls and running tests twice in a row which naturally failed